### PR TITLE
most feedback changes + incorporate expansion char/mapshown fixes from 141

### PR DESCRIPTION
**Pull request by RushTheOne**

### CODE REVIEW

File: Types/NPC.cs [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+**/
+using System.Collections.Generic;
 namespace MapAssist.Types
 {
+    public static class NPC
+    {
+        public static HashSet<uint> Dummies = new HashSet<uint> { 149, 151, 152, 153, 157, 158, 159, 195, 196, 197, 179, 185, 203, 204, 205, 268, 269, 271, 272, 293, 294, 289, 290, 291, 292, 296, 318, 319, 320, 321, 322, 323, 324, 325, 332, 338, 339, 344, 355, 359, 363, 364, 370, 377, 378, 392, 393, 401, 411, 412, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 711 };
```
Comment (misterokaygo) [inline Types/NPC.cs:24]: I know this is tedious but can we convert the Dummies to the enum Npc and make this HashSet use the Npc Enums

Npc.Dummy1 ,etc

File: Types/UnitAny.cs [hunk 2]
```diff
@@ -105,21 +108,33 @@ public bool IsPlayer()
 
         public bool IsPlayerUnit()
         {
-            return IsPlayer() && _unitAny.pInventory != IntPtr.Zero && Inventory.pUnk1 != IntPtr.Zero;
+            using (var processContext = GameManager.GetProcessContext())
```
Comment (misterokaygo) [inline Types/UnitAny.cs:111]: Future note, this kind of static is non-testable. Thinking we should wrap it an instance method in a future PR.

File: Types/UnitAny.cs [hunk 3]
```diff
@@ -105,21 +108,33 @@ public bool IsPlayer()
...
             if (_unitAny.Mode == 0 || _unitAny.Mode == 12) return false;
-            if ((_unitAny.TxtFileNo >= 110 && _unitAny.TxtFileNo <= 113) || (_unitAny.TxtFileNo == 608 && _unitAny.Mode == 8)) return false;
-            if (_unitAny.TxtFileNo == 68 && _unitAny.Mode == 14) return false;
-            if ((_unitAny.TxtFileNo == 258 || _unitAny.TxtFileNo == 261) && (_unitAny.Mode == 14)) return false;
-            // if (D2COMMON_get_unit_stat(unit, 172, 0) == 2) return 0;
-            var falsePositives = new List<uint> { 227, 283, 326, 327, 328, 329, 330, 410, 411, 412, 413, 414, 415, 416, 366, 406, 351, 352, 353, 266, 408, 516, 517, 518, 519, 522, 523, 543, 543, 545 };
-            if (falsePositives.Contains(_unitAny.TxtFileNo)) return false;
-            // ms_wchar_t* wname = D2CLIENT_get_unit_name(unit);
-            // if ((strcmp(name, "an evil force") == 0) || (strcmp(name, "dummy") == 0))
+            if (NPC.Dummies.Contains(_unitAny.TxtFileNo)) { return false; }
```
Comment (misterokaygo) [inline Types/UnitAny.cs:137]: 👍

**Review by misterokaygo — APPROVED** [review]

File: Helpers/GameManager.cs [hunk 4]
```diff
@@ -40,8 +40,15 @@ public static ProcessContext GetProcessContext()
...
             {
-                _ProcessContext.OpenContextCount++;
-                return _ProcessContext;
+                IntPtr windowInFocus = WindowsExternal.GetForegroundWindow();
+                if(_MainWindowHandle == windowInFocus)
+                {
+                    _ProcessContext.OpenContextCount++;
+                    return _ProcessContext;
+                } else
+                {
```
Comment (AudriusButkevicius) [inline Helpers/GameManager.cs:49]: the formatting looks wonky
no space after if and wierd line breaks on else.

File: Types/NPC.cs (hunk 1 above)
Comment (AudriusButkevicius) [reply Types/NPC.cs:24]: Agreed. Having the names of the actual dummies would be useful.

File: Types/UnitAny.cs [hunk 5]
```diff
@@ -105,21 +108,33 @@ public bool IsPlayer()
...
+                {
+                    var expansionCharacter = processContext.Read<byte>(processContext.FromOffset(Offsets.ExpansionCheck)) == 1;
+                    var userBaseOffset = 0x30;
+                    var checkUser1 = 1;
+                    if (expansionCharacter)
+                    {
+                        userBaseOffset = 0x70;
+                        checkUser1 = 0;
+                    }
+                    var UserBaseCheck = processContext.Read<int>(IntPtr.Add(_unitAny.pInventory, userBaseOffset));
```
Comment (AudriusButkevicius) [inline Types/UnitAny.cs:123]: should be lowercase?

**Review by AudriusButkevicius — APPROVED** [review]

**Pull request merged**